### PR TITLE
Adding async search to 2.3 manifest 

### PR DESCRIPTION
**Pull request by bharath-techie**
### Description
Adding async search to 2.3 manifest

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/2.3.0/opensearch-2.3.0.yml [hunk 1]
```diff
@@ -99,3 +99,9 @@ components:
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
+  - name: asynchronous-search
+    repository: https://github.com/opensearch-project/asynchronous-search.git
+    ref: '2.3'
```
Comment (prudhvigodithi) [inline manifests/2.3.0/opensearch-2.3.0.yml:104]: Hey @bharath-techie I dont see 2.3 branch created for `asynchronous-search`, can you please check?

Comment (bharath-techie) [reply manifests/2.3.0/opensearch-2.3.0.yml:104]: Hi, now I've created, can you please retrigger builds again

Comment (prudhvigodithi) [reply manifests/2.3.0/opensearch-2.3.0.yml:104]: Hey @bharath-techie thanks, tests now pass, but there arise some conflict.

Comment (gaiksaya) [reply manifests/2.3.0/opensearch-2.3.0.yml:104]: Please add `platform` as well. Also duplicate PR #2585 ?

**Comment by gaiksaya** [comment]
Closing this as the other one was merged.

**Pull request closed without merging**